### PR TITLE
Server fixes

### PR DESCRIPTION
**Pull request by ajacksified**
:eyeglasses: @schwers or @curioussavage

### CODE REVIEW

File: src/routes.jsx [hunk 1]
```diff
@@ -52,12 +52,16 @@ function buildAPIOptions(ctx, options={}) {
...
-    },
+    headers: { },
     env: ctx.env || 'SERVER',
   }, options);
 
+  if (ctx.headers['user-agent']) {
+    apiOptions.headers['user-agent'] = ctx.headers['user-agent'];
+  }
+
+  console.log(apiOptions.headers['user-agent']);
```
Comment (schwers) [inline src/routes.jsx]: is this needed for debug?

File: src/views/pages/index.jsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ class IndexPage extends BasePage {
     subredditName: React.PropTypes.string,
   };
 
-  static isStalePage (query, listings) {
+  static isStalePage (query, listings={}) {
```
Comment (schwers) [inline src/views/pages/index.jsx]: Do you have a link that repros the undefined error? `listings` was `undefined` before so shouldn't `!headers.before` through a `can't read property 'before' of undefined` error?

Comment (ajacksified) [inline src/views/pages/index.jsx]: I don't have a link - I tested it by setting `this.props.datacache.listings = undefined` in routes in the line before isStalePage is called. Oddly.. it didn't throw, but it looks like it should have. I'l just return false early if `listings` is undefined.

Comment (schwers) [inline src/views/pages/index.jsx]: :+1:

Comment (schwers) [inline src/views/pages/index.jsx]: That is odd though, (shrug)

**Comment by schwers** [comment]
:fish:

**Pull request merged**